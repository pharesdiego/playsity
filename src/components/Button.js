import React from 'react';
import { Flex } from './Styles';

const Container = Flex.withComponent('button').extend`
  width: auto;
  height: auto;
  color: ${props => props.color || 'grey'};
  padding: 10px 20px;
  box-shadow: 0px 0px 10px 2px rgba(62, 62, 62, 0.30);
  border-radius: 4px;
  border: 1px solid rgba(255, 255, 255, 0.26);
  background-color: ${props => props.bg || 'transparent'};
`;

const Button = ({ children, color, bg, ...rest}) => {
  return (
    <Container color={color} bg={bg} {...rest}>
      { children }
    </Container>
  )
}

export default Button;