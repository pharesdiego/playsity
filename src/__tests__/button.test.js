import React from 'react';
import Button from './../components/Button';

const wrapper = mount(<Button color="white" bg="black"/>);
describe('>>> <Button> <<<', () => {
  it('Must render', () => {
    expect(wrapper).toBeDefined();
  });

  it('Must render properly with passed props', () => {
    expect(wrapper.find('Button').props()).toMatchSnapshot();
  });
});