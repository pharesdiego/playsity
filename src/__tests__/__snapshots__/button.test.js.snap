// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> <Button> <<< Must render properly with passed props 1`] = `
Object {
  "bg": "black",
  "color": "white",
}
`;
